### ISSUE TITLE
`req.params().get("name")` does not return a value while `req.params("name")` does

### ISSUE THREAD

**Issue by bentolor** [perwendel/spark#339]
Spark maintains a unquoted map of path parameter names to values. So the map returned by `req.params()` contains key/values in the form `:key,value`, while the `params("name")` without the colon is happily accepted. See the implementation code below.

Besides not being documented, in my humble opinion this totally works against [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) and Spark should return a map with `key,value`-pairs instead of `:key,value`.

This would allow things like: 

     get("/greet/:name", 
           (req,res) -> new ModelAndView(req.params(), "greet"), new JadeTemplateEngine())

which currently do not work

What is the rationale for the current `params(name)`-method to look like
```
    public String params(String param) {
        if (param.startsWith(":")) {
            return params.get(param.toLowerCase());
        } else {
            return params.get(":" + param.toLowerCase());
        }
    }
```

instead of:

```
    public String params(String param) {
            return params.get(param);
    }
```

**Comment by bentolor** [comment]
Also: The behavior is also contradictory to `req.queryMap()`, which also returns a map with plain names.

**Comment by perwendel** [comment]
The rationale was that it should work as it does in Sinatra, and the logic:

The colon indicates that it's a parameter and then comes the parameter name which you later use for getting the parameter value. I don't think that is against POLA.

However, the result from `params()` is strange and as you point out it should have returned a list with  `name, value` pairs. The problem is that this has been the implementation since day 1 and changing might have nasty side-effects for a lot of users which they would catch in runtime since no compilation would be broken.

**Comment by rguderlei** [comment]
For me, the ":key" thing looks like a ruby symbol. Maybe it was the intention to emulate ruby symbols? I.e. in ruby `params[:name]` and `params["name"]` are equivalent, whereas `params[":name"]` does not make sense (and would not work in Sinatra AFAIK).

**Comment by bentolor** [comment]
I agree that the usage of the colon within the URL pattern is totally the way someone expects it to work. As in in Sinatra the colon should does not make any sense to appear anywhere in the API. The Ruby symbol notation allowing `params[:name]` is a language built-in concept, not an API feature. 

I completely understand the backwards-compatibility argument. But I'd even dare to claim that the current behavior is so confusing, that it _should_ be changed & announced even if this _could break_ existing code. `params` itself should just behave the same way as in Sinatra.

The typical way to handle this by adding an alternative method like `params2()`  and `@Deprecated` the original one is IMHO not appropriate for a microframework like Spark.

**Comment by nicetryzz** [comment]
I would like to fix this bug. It need to change the getParams  and params function in request class, which guarantee the key in the map won't start with ":".
